src/app/components/SearchResults/Filters.jsx

```diff
@@ -0,0 +1,139 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { initialSearchQuery, searchQueryPropTypes } from '../../stores/InitialState';
+import { getQueryString } from '../../search/query';
+
+const filtersLabels = { language: 'Language' };
+
+const Filters = ({
+  results, searchQuery, userQuery, router,
+}) => {
+  const filtersArray = [];
+
+  // add search filters
+  if (searchQuery && searchQuery.filters) {
+    searchQuery.filters.forEach((filter) => {
+      filtersArray.push({ field: filter.field, value: filter.value });
+    });
+  }
+
+  // redirect to url with query params
+  const submit = (query) => {
+    const path = `/search?${getQueryString(query)}`;
+    router.push(path);
+  };
+
+  // update page in store and go to any page
+  const doSearchWithFilters = (filters) => {
+    const newQuery = Object.assign({}, searchQuery, { filters }, { page: 0 });
+    userQuery(newQuery);
+    submit(newQuery);
+  };
+
+  // on ckech of filter, add it or remove it from list and do the search
+  const onChangeCheckbox = (e, field, value) => {
+    const matchIndex = filtersArray.findIndex(filter => filter.field === field && filter.value === value);
+    if (e.target.checked && matchIndex === -1) {
```

Comment:
Is the `e.target.checked` necessary here? I only ask because it seems a bit counter-intuitive to see if the checkbox is checked to verify that we should be adding this as a new filter, at first look I thought it meant that it had already been checked and should be removed

Reply:
Yes, as it also removes the filter when it's unchecked